strapi/strapi
# fix: cancel button not respected on image deletion

### What does it do?
Fixes [issue](https://github.com/strapi/strapi/issues/17786) with cancel button deleting assets rather than just closing the dialog, the solution introduces a refinement in the `onClose` prop by modifying it to accept a parameter named `value`. This parameter corresponds to the value that indicates the user's choice: whether to proceed with the `deletion` or to `cancel` the operation. By incorporating this parameter, the solution enables the dialog to differentiate between the two scenarios.

### Why is is needed?
To prevent user from deleting assets (image) when they have instead chosen to cancel their edition.

### [Fixes/issue](https://github.com/strapi/strapi/issues/17786)

### How to test it?
1. Create a `Content-Type` that accepts an asset (image) and save.
2.  In Content Manager, add an asset.
3.  Click on the edit icon.
4. Click on the delete icon on the details modal.
5. Now click the cancel button instead when prompted if you want to delete.
6. See that the asset (image) is not deleted on cancellation.

### Demo

[Loom Video](https://www.loom.com/share/75d72aa49901482697053bca383824f0?sid=ccfaacea-d7c8-4400-9992-057d4ea53761)
